base/ltoutput.dtx

```diff
@@ -1213,6 +1232,240 @@
+%      decided which floats go into which area and which get deferred.
+%      Therefore, the assumption is that the code in the socket attaches all
+%      areas that contain floats. If this is not done, then the order
+%      of floats is likely to be screwed up unless unused floats
+%      are moved to the defer list in an appropriate way (for now we
+%      don't offer any interface for that scenario). 
+%
+%      Before the code in the socket is run, any existing glue at the bottom
+%      of the \cs{@outputbox} is removed and stored in a safe
+%      place. If needed, it can be reinserted with one of the helper
+%      commands.
+%
+%      To support setting this up the following helper commands are available:
+%      \begin{description}
+%      \item[\cs{@outputbox@append} (1 argument)]
+%
+%        \DescribeMacro[noprint]{@outputbox@append}
+%
+%        This general purpose command alters the \cs{@outputbox} box by
+%        appending material to it.
+%
+%      \item[\cs{@outputbox@appendfootnotes} (0 arguments)]
+%
+%        \DescribeMacro[noprint]{@outputbox@appendfootnotes}
+%
+%        This command appends the footnotes to the \cs{@outputbox} (if
+%        there are any). If not, then it does nothing.
+%
+%      \item[\cs{@outputbox@attachfloats} (0 arguments)]
+%      \item[\cs{@outputbox@attachtopfloats} (0 arguments)]
+%      \item[\cs{@outputbox@attachbottomfloats} (0 arguments)]
+%
+%        \DescribeMacro[noprint]{@outputbox@attachfloats}
+%        \DescribeMacro[noprint]{@outputbox@attachtopfloats}
+%        \DescribeMacro[noprint]{@outputbox@attachbottomfloats}
+%   
+%        Attaching top and bottom floats can usually be done in one
+%        go, but for special layouts we might want more control so we
+%        provide also separate commands.
+%
+%      \item[\cs{@outputbox@reinsertbskip} (0 arguments)]
+%
+%        \DescribeMacro[noprint]{@outputbox@reinsertbskip}
+%
+%        Reinsert the bottom skip of the \cs{@outputbox} that was
+%        saved before in \cs{@makecol}.   
+%   
+%      \item[Testing for existence of material]
+%
+%        \DescribeMacro[noprint]{@if@footnotes@TF}
+%        \DescribeMacro[noprint]{@if@bfloats@TF}
+%
+%        There are a number of helpers to run conditional code
+%        depending on whether or not there are footnotes or bottom
+%        floats. They are \cs{@if@footnotes@TF} and
+%        \cs{@if@bfloats@TF}
+%        (names are likely to change).
+%   
+%      \end{description}
+%
+%      \task{fmi}{Decide on names for these helper commands. We could keep
+%        them similar to the above (because they are only supposed to be
+%        used by a few packages). However, we could alternatively use
+%        CamelCase names, under the assumption that classes may also
+%        directly define a plug for the socket, instead of loading a
+%        support package such as \pkg{footmisc} (or in addition
+%        to). --- decide}
```

Comment:
If we are going with the current names it's probably best to delete this `\task` as suggesting the names will change will discourage use.

Reply:
Well, I'm just not sure if the assumption is really correct. and there will be any usage outside of things like footmisc or ftnright and there the nams with `@` are probably fine. The task is there mainly to suggest that people think if there are use cases.